pkg/cluster/internal/kubeadm/config.go

```diff
@@ -621,6 +627,13 @@ func Config(data ConfigData) (config string, err error) {
 		data.KubeadmFeatureGates["IPv6DualStack"] = true
 	}
 
+	// before 1.24 kind uses cgroupfs
+	// after 1.24 kind uses systemd starting in kind v0.13.0
+	// before kind v0.13.0 kubernetes 1.24 wasn't released yet
+	if ver.LessThan(version.MustParseSemantic("v1.24.0")) {
+		data.CgroupDriver = "cgroupfs"
```

Comment:
this effectively means that  `ConfigTemplateBetaV1` and `ConfigTemplateBetaV2` are going to use data.CgroupDriver = "cgroupfs", maybe for consistency is better to replace in all templates?

Reply:
Do you mean don't template it?

I don't think we should do that no, selecting based on Kubernetes version is the current approach and this is more explicit about how that is selected. Templating it everywhere ensures greater logical consistency.